addons/rook-ceph/enable

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+import click
+import os
+import re
+import subprocess
+import shutil
+from pathlib import Path
+
+# Script dir
+DIR = Path(__file__).absolute().parent
+
+# Snap paths
+SNAP = Path(os.getenv("SNAP") or "/snap/microk8s/current")
+SNAP_COMMON = Path(os.getenv("SNAP_COMMON") or "/var/snap/microk8s/common")
+
+# Snap tools and known paths
+GIT = SNAP / "git.wrapper"
+KUBECTL = SNAP / "microk8s-kubectl.wrapper"
+KUBELET_DIR = SNAP_COMMON / "var" / "lib" / "kubelet"
+
+# Addon specifics
+ROOK = SNAP_COMMON / "plugins" / ".rook"
+ROOK_RESOURCES = ["crds.yaml", "common.yaml", "operator.yaml"]
+ROOK_PLUGINS = [
+    "import-ceph-cluster",
+    ".rook-create-external-cluster-resources.py",
+    ".rook-import-external-cluster.sh",
+]
+
+
+def patch_operator_kubelet_dir(path: Path):
+    """Patch the kubelet dir in operator.yaml"""
+    match = r"(# )?ROOK_CSI_KUBELET_DIR_PATH.*"
+    replace_with = f"ROOK_CSI_KUBELET_DIR_PATH: {KUBELET_DIR}"
+
+    contents = path.read_text()
+    new_contents = re.sub(match, replace_with, contents)
+
+    if contents != new_contents:
+        path.write_text(new_contents)
+
+
+@click.command()
+@click.option("--rook-version", default="v1.11.9")
+@click.option("--rook-repository", default="https://github.com/rook/rook")
+def main(rook_version: str, rook_repository: str):
+    if ROOK.exists():
+        shutil.rmtree(ROOK, ignore_errors=True)
+
+    click.echo(f"Fetch {rook_repository}@{rook_version}")
+    subprocess.check_call([GIT, "clone", rook_repository, "--depth=1", ROOK])
```

Comment:
Could we deploy this via helm?

Reply:
The biggest issue with the Helm charts is that the CRD handling can be tricky when upgrading to a newer Rook version. Specifically, it is possible for the chart option to deploy the CRDs to be missed by accident, which would cause the Rook CRDs to be deleted from the cluster (causing catastrophic data loss)

If that is an acceptable risk and is properly documented (like Rook does), then happy to adjust

Reply:
Updated to use helm charts